### PR TITLE
FEATURE: BIA-534 Allow max retry override for tests via ENV variable

### PR DESCRIPTION
**Pull request by vhtien**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/40/ : FAILURE in 23s
- gooddata-ruby-tests-zuul : SKIPPED

File: spec/unit/rest/polling_spec.rb [hunk 1]
```diff
@@ -75,7 +75,7 @@
 
   it 'should make 1 additional request when hitting 500' do
     expect do
-      @client.get('/internal_error')
+      @client.get('/internal_error', :tries => 3)
```
Comment (korczis) [inline spec/unit/rest/polling_spec.rb]: Why there is explicit :tries => 3, is not default sufficient?

Comment (vhtien) [inline spec/unit/rest/polling_spec.rb]: because of https://github.com/gooddata/gooddata-ruby/pull/701/files#diff-b3fa27bc6eddfdd4e184fd694324e069R80

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/41/ : FAILURE in 22s
- gooddata-ruby-tests-zuul : SKIPPED

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/42/ : FAILURE in 23s
- gooddata-ruby-tests-zuul : SKIPPED

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/43/ : FAILURE in 14s
- gooddata-ruby-tests-zuul : SKIPPED

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/44/ : FAILURE in 24s
- gooddata-ruby-tests-zuul : SKIPPED

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/45/ : SUCCESS in 24s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/42/ : FAILURE in 1m 30s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/46/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/43/ : FAILURE in 1m 28s

**Comment by vhtien** [comment]
ok to test

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/47/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/44/ : FAILURE in 1m 27s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/48/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/45/ : FAILURE in 1m 19s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/49/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/46/ : FAILURE in 1m 22s

**Pull request merged**